src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractJenkinsMojo.java

```diff
@@ -89,6 +94,39 @@ protected String findJenkinsVersion() throws MojoExecutionException {
         throw new MojoExecutionException("Failed to determine Jenkins version this plugin depends on.");
     }
 
+    protected JavaSpecificationVersion getMinimumJavaVersion() throws MojoExecutionException {
+        Artifact bom = resolveJenkinsCoreBom(findJenkinsVersion());
+
+        Properties properties;
+        try {
+            properties = wrap(bom).resolvePom().getProperties();
+        } catch (ProjectBuildingException e) {
+            throw new MojoExecutionException("Failed to resolve artifact " + bom, e);
+        }
+
+        String version = properties.getProperty("java.level");
+        if (version == null || version.isEmpty()) {
+            throw new MojoExecutionException("java.level not defined in " + bom);
+        }
```

Comment:
This is probably the most controversial part of this change, as it commits us to an encoding of `java.level` in the `<properties>` section of `core-bom`. Might not be the ideal encoding. As I wrote above, my point here is just to rely on an encoding _somewhere_. It doesn't have to be this specific encoding.

Reply:
Can we inspect the Java bytecode level of some class in `jenkins-core.jar`? More work, but less sensitive to how core is built.

Reply:
e.g. https://stackoverflow.com/a/1707200/12916 / https://stackoverflow.com/a/1293372/12916

Reply:
That's an interesting idea. I'll try it out. I like it on a high level. As the late Roger Faulkner once opined:

> Roger believed that terrible things were sometimes required to create beautiful abstractions, and his trailblazing work on `/proc` embodies this burden: the innards may be delicate and nasty (‘vile,’ as Roger might say in his distinguished Carolinian accent) — but the resulting abstractions are breathtaking in their power, scope and robustness.

Reply:
Done in commit 34cf73b9736fb1b619bc148f5a9fca45528bb610. Marking as draft due to the incremental in the POM file, but from my perspective this is ready for review, just not ready for merge until the release of `lib-version-number`.